### ISSUE TITLE
require-array-sort-compare incorrectly warns on a enum string array

### ISSUE THREAD

**Issue by MattisAbrahamsson** [oxc-project/tsgolint#539]
require-array-sort-compare will by default not warn on string arrays being sorted without a compare function, however if the string is an enum oxlint will generate a warning

```
declare const myArray: Array<"A" | "B">;

myArray.sort();
```

typescript-eslint does not generate a warning on the code above
